### ISSUE TITLE
Cannot pass Table reference to row() constructor in array or list

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15008]
Passing a `Table` reference to a `row()` constructor that accepts a `Collection<?>` or `SelectField<?>...` results in this exception: 

```
java.lang.IllegalArgumentException: Cannot interpret argument of type class org.jooq.test.postgres.generatedclasses.public_.tables.TLanguage as a Field: "public"."t_language"
	at org.jooq.impl.Tools.fieldExpected(Tools.java:1831)
	at org.jooq.impl.Tools.field(Tools.java:2079)
	at org.jooq.impl.Tools.field(Tools.java:2087)
	at org.jooq.impl.Tools.lambda$64(Tools.java:2108)
	at org.jooq.impl.Tools.map(Tools.java:2330)
	at org.jooq.impl.Tools.fieldsArray(Tools.java:2108)
	at org.jooq.impl.DSL.row(DSL.java:32721)
	at org.jooq.impl.DSL.row(DSL.java:33037)
	at org.jooq.impl.DSL.multisetAgg(DSL.java:27270)
	at org.jooq.test.all.testcases.MultisetTests.lambda$21(MultisetTests.java:475)
	at org.jooq.test.jOOQAbstractTest.withDSLContext(jOOQAbstractTest.java:2340)
	at org.jooq.test.jOOQAbstractTest.bindAndInlineWithDSLContexts(jOOQAbstractTest.java:2333)
	at org.jooq.test.BaseTest.bindAndInlineWithDSLContexts(BaseTest.java:2031)
	at org.jooq.test.all.testcases.MultisetTests.lambda$20(MultisetTests.java:446)
	at org.jooq.test.all.testcases.MultisetTests.lambda$538(MultisetTests.java:3456)
	at org.jooq.test.all.testcases.MultisetTests.lambda$541(MultisetTests.java:3474)
	at org.jooq.test.all.testcases.MultisetTests.testRecordLevelListenerResults(MultisetTests.java:3513)
	at org.jooq.test.all.testcases.MultisetTests.testRecordLevelListenerResults(MultisetTests.java:3501)
	at org.jooq.test.all.testcases.MultisetTests.testMultisetEmulations(MultisetTests.java:3474)
	at org.jooq.test.all.testcases.MultisetTests.testMultisetEmulations(MultisetTests.java:3456)
	at org.jooq.test.all.testcases.MultisetTests.testMultisetAggNestedTableAsFields(MultisetTests.java:445)
	at org.jooq.test.jOOQAbstractTest.testMultisetAggNestedTableAsFields(jOOQAbstractTest.java:5668)
```

The 22-degree constructors don't have this problem. I.e.:

```java
// Works
row(TABLE)

// Doesn't work
row(List.of(TABLE))
```

**Comment by lukaseder** [comment]
This issue was discovered when writing a test for https://github.com/jOOQ/jOOQ/issues/15007

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0, 3.18.4 (https://github.com/jOOQ/jOOQ/issues/15009), and 3.17.13 (https://github.com/jOOQ/jOOQ/issues/15010)
